servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.kt

```diff
@@ -60,45 +62,52 @@ const val DEFAULT_INSTRUMENTATION_ORDER = 0
     FederatedSchemaAutoConfiguration::class
 )
 class GraphQLSchemaConfiguration {
-
     @Bean
     @ConditionalOnMissingBean
     fun graphQL(
         schema: GraphQLSchema,
         dataFetcherExceptionHandler: DataFetcherExceptionHandler,
-        instrumentations: Optional<List<Instrumentation>>,
+        providedInstrumentations: Optional<List<Instrumentation>>,
         executionIdProvider: Optional<ExecutionIdProvider>,
         preparsedDocumentProvider: Optional<PreparsedDocumentProvider>,
         config: GraphQLConfigurationProperties,
         idValueUnboxer: IDValueUnboxer
     ): GraphQL {
-        val graphQL = GraphQL.newGraphQL(schema)
+        val graphQLBuilder = GraphQL.newGraphQL(schema)
             .queryExecutionStrategy(AsyncExecutionStrategy(dataFetcherExceptionHandler))
             .mutationExecutionStrategy(AsyncSerialExecutionStrategy(dataFetcherExceptionHandler))
             .subscriptionExecutionStrategy(FlowSubscriptionExecutionStrategy(dataFetcherExceptionHandler))
             .valueUnboxer(idValueUnboxer)
+            .also { builder ->
+                executionIdProvider.ifPresent(builder::executionIdProvider)
+                preparsedDocumentProvider.ifPresent(builder::preparsedDocumentProvider)
+
+                val instrumentations = mutableListOf<Instrumentation>()
+                if (config.batching.enabled) {
```

Comment:
add batch instrumentations if batching is enabled, otherwise stick to the default behavior which is adding the `DefaultDataLoaderDispatcherInstrumentation` from graphql-java, which applies to a single operation